### ISSUE TITLE
Download of a file larger than 2 GB

### ISSUE THREAD

**Issue by shimon128** [DigitalMediaServer/NSIS-INetC-plugin#10]
I trying to download a file whose size is about 12.7 GB but it constantly lowers me only 692 MB.
What can I do?
![image](https://user-images.githubusercontent.com/70193333/224552140-56613617-5285-49e6-b4c0-287a01926987.png)

**Comment by Nadahar** [comment]
It should download the full file - it's just the display that is wrong. It's because of an overflow in the int holding the size value - but I haven't been able to figure out how to get a larger size to use because I don't get the type system and the limitations for what types can be used and not.

**Comment by Nadahar** [comment]
Have you tried to use [NSCurl](https://github.com/negrutiu/nsis-nscurl) instead?

**Comment by shimon128** [comment]
Yes, I tried it.
At the moment I have other problems with it that for some reason the download speed starts start great but then drops drastically.
Are you sure it's just showing the wrong size? Because it ends very quickly.

**Comment by Nadahar** [comment]
No, I'm not sure - I just thought that was the case. The actual download is done by the Internet Explorer part of Windows of all things, so maybe the Internet Explorer version installed might impact it.

To be honest, that's the problem with this plugin, that it's based on `MS WinInet API`, which is again using Internet Explorer behind the scenes. Since IE is hopelessly outdated and have been abandoned for a long time, this plugin suffers as well. That's why I think a curl based plugin sounds like a much better idea in 2023.

**Comment by Nadahar** [comment]
I haven't written the plugin itself, I've just modified the dialog box. I'm not really a C/C++ developer, which is why it doesn't take much for me to get in deep water. This is based on the "official" INetC plugin, so maybe they can address the issue. If I know the fix, I can implement it here as well.
